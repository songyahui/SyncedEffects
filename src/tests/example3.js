"use hopscript"

var hh = require("hiphop");

var prg = <hh.module A=${{accessibility: hh.IN}} T V>
  <hh.abort A>
    <hh.local S>
      <hh.loop>
	<hh.sequence>
	  <hh.emit S/>
	  <hh.if S>
	    <hh.emit T/>
	  </hh.if>
	  <hh.pause/>
	  <hh.emit V/>
	</hh.sequence>
      </hh.loop>
    </hh.local>
  </hh.abort>
</hh.module>

exports.prg = new hh.ReactiveMachine(prg, "example3");
