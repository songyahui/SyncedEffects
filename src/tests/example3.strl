module example3:

input A;
output T, V;

signal S in
       abort
	       loop
	       		emit S;
	       		present S then emit T end;
	       		pause;
	       		emit V
	       end loop
       when A
end signal

end module