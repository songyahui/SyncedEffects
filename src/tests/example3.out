example3> ;
--- Output: T
example3> ;
--- Output: T V
example3> ;
--- Output: T V
example3> ;
--- Output: T V
example3> ;
--- Output: T V
example3> A;
--- Output:
example3> ;
--- Output:
example3> ;
--- Output:
example3> ;
--- Output:
example3> ;
--- Output:
